src/features/beta-feedback/BetaFeedbackBadge.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import classNames from 'classnames';
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import { BetaBadge } from '../../components/badge';
+import { Link } from '../../components/link';
+import Tooltip from '../../components/tooltip/Tooltip';
+
+import './styles/BetaFeedbackBadge.scss';
+
+type Props = {
+    className?: string,
+    formUrl: string,
+    tooltip?: boolean,
+};
+
+const BetaFeedbackBadge = ({ className = '', tooltip = false, ...rest }: Props) => {
+    const classes = classNames('bdl-HeaderFeedbackBadge', className);
+    const { formUrl } = rest;
+    const badge = tooltip ? (
+        <Tooltip
+            text={
+                <FormattedMessage
+                    id="be.feedbackCtaText"
+                    defaultMessage="Click to provide feedback"
+                    description="Call-to-action text describing what to do to navigate to specified feedback form"
+                />
+            }
+            position="middle-right"
+        >
+            <BetaBadge aria-hidden className="bdl-HeaderFeedbackBadge-betaBadge" />
+        </Tooltip>
+    ) : (
+        <BetaBadge className="bdl-HeaderFeedbackBadge-betaBadge" />
+    );
+
+    // TODO: tooltip may require constrainToScrollParent & constrainToWindow in some contexts
+    return (
+        <span className={classes}>
+            <span id="bdl-HeaderFeedbackBadge-ariaLabel" aria-hidden="true" hidden>
+                <FormattedMessage
+                    id="be.feedbackFormDescription"
```

Comment:
To avoid writing these out you can `import messages from '../messages';`, and then spread the object into the `FormattedMessage` component. 

`<FormattedMessage {...messages. feedbackFormDescription} />`

Reply:
I am starting to lean towards using FormattedMessage inline:
- Because defineMessages returns a new object, there is no way to easily jump to the variable definition for a message, nor to easily find usages of a message. For that reason I think there a lot of dead messages that don't show up in any components any more.
- messages.js is an indirection in the best case

Reply:
ah yeah, initially there was only one message to format, so wanted to avoid creating a whole module for one string. if we had 3+, I'd convert over to a messages module

Reply:
@alexkrolick we can describe some rules around it, and I think there's a case for both, given a few criteria. For instance, things like:

- well-defined strings that are used across many components, versus component-specific strings
- number of strings defined for a specific component
- likelihood that a message may become obsolete

Reply:
I don’t see any other examples of inline FormattedMessage components in Elements. Can you move the definition into a messages module to stay consistent with the rest of the project?

Reply:
https://github.com/box/box-ui-elements/pull/1037 turns out Flow can work with jump-to-definition if you don't cast the result of `defineMessages`